### PR TITLE
Update DatePicker

### LINKED ISSUE #29: [Feature] Datepicker (Experimental)
# Experimental

[Date picker component](https://flowbite.com/docs/plugins/datepicker/).

- [x] Default
- [ ] Inline date picker
- [x] Date range picker
- [ ] Autohide
- [x] Action buttons
- [x] Date format
- [x] Orientation
- [x] Title

### LINKED ISSUE #1216: DatePicker enchancements
## Summary

A couple of features for the date picker:

1.) Set starting date of the calendar
2.) Set initial selected date

### Motivation

It would be nice to be able to set dates on the calendar.

### LINKED ISSUE #1283: Datepicker
When do you think we'll see a datepicker promoted out of experimental?

`⚠️ Datepicker component is experimental and only works with SvelteKit.`

### PR DESCRIPTION
**Pull request by Chizaruu**
Closes #1283
Closes #1216
Closes #29 

## 📑 Description

This pull request is to address the DataPicker issues.

Please see DatePicker docs in PR to view changes and examples.

## Status

- [x] Completed

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [ ] I have checked the page with https://validator.unl.edu/
- [ ] All the tests have passed
- [x] My pull request is based on the latest commit (not the npm version).

## ℹ Additional Information

Let me know if I need to add anything.

## Summary by CodeRabbit

- **New Features**
	- Introduced a revamped DatePicker component with enhanced customization options and improved user experience.
	- Added comprehensive documentation for the DatePicker component, detailing setup, usage, and customization.

- **Bug Fixes**
	- Removed outdated components and properties to streamline functionality.

- **Refactor**
	~~- Updated component naming conventions for consistency and clarity, including renaming the Datepicker to DatePicker.~~ Reversed to Datepicker.

- **Chores**
	- Cleaned up obsolete files related to the previous date picker implementation.

### CODE REVIEW

File: src/routes/component-data/Datepicker.json [hunk 1]
```diff
@@ -1,14 +1,24 @@
 {
-  "name": "Datepicker",
+  "name": "DatePicker",
```
[the remark this answers is not in the document]
Comment (Chizaruu) [reply src/routes/component-data/Datepicker.json]: Changed back to Datepicker

**Comment by shinokada** [comment]
Thank you for the PR.

Is Required Field supposed to have any sign or symbol for the required?
image

**Comment by shinokada** [comment]
- fixed test fails
- fixed `pnpm check` for `focusedDate` and `isSelected` functions
- update eslint.config.js
- postcss.config.js
- I will update later for `pnpm format && pnpm lint` errors

**Comment by shinokada** [comment]
Announcement at discord, https://discord.com/channels/902911619032576090/928736443218665563.

image

**Comment by Chizaruu** [comment]
> Thank you for the PR.
> 
> Is Required Field supposed to have any sign or symbol for the required?

While adding one is possible, I don't know if we should, as the developer may want to use the Label component for the sign or symbol.

**Comment by shinokada** [comment]
Yes, I agree with you that developer can add it to the label.
Thanks.

**Comment by shinokada** [comment]
FYI: I changed flowbite-svelte-icons to svg icons in order not to add another dependency to the library. Some uses are using another icon libraries, https://svelte-svg-icons.codewithshin.com/.

**Comment by shinokada** [comment]
Can you have a look at https://github.com/themesberg/flowbite-svelte/pull/1476 ?

**Pull request merged**